File: fix_helmet_imports.py
#!/usr/bin/env python3
"""
Script to fix missing Helmet imports in TSX files.
"""

import os
import re
from pathlib import Path

def fix_helmet_imports_in_file(file_path):
    """Fix missing Helmet imports in a single file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file uses Helmet but doesn't import it
        if 'Helmet' in content and 'react-helmet-async' not in content:
            print(f"Fixing Helmet import in: {file_path}")
            
            # Find the first import line
            lines = content.split('\n')
            import_line_index = -1
            
            for i, line in enumerate(lines):
                if line.strip().startswith('import '):
                    import_line_index = i
                elif line.strip() and not line.strip().startswith('//') and not line.strip().startswith('/*'):
                    break
            
            # Add the Helmet import
            if import_line_index >= 0:
                lines.insert(import_line_index + 1, "import { Helmet } from 'react-helmet-async';")
            else:
                lines.insert(0, "import { Helmet } from 'react-helmet-async';")
            
            # Write the fixed content back
            fixed_content = '\n'.join(lines)
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(fixed_content)
            
            return True
        
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix Helmet imports."""
    workspace_root = Path('/workspace')
    app_dir = workspace_root / 'app'
    
    if not app_dir.exists():
        print("App directory not found!")
        return
    
    # Find all TSX files
    tsx_files = list(app_dir.rglob('*.tsx'))
    
    print(f"Found {len(tsx_files)} TSX files to check")
    
    fixed_count = 0
    for file_path in tsx_files:
        if fix_helmet_imports_in_file(file_path):
            fixed_count += 1
    
    print(f"Fixed Helmet imports in {fixed_count} files")

if __name__ == "__main__":
    main()

File: fix-helmet-imports.js
import fs from 'fs';
import path from 'path';

function updateFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Replace react-helmet-async imports with next/head
    content = content.replace(
      /import { Helmet } from 'react-helmet-async';/g,
      "import Head from 'next/head';"
    );
    content = content.replace(
      /import { Helmet } from "react-helmet-async";/g,
      'import Head from "next/head";'
    );
    
    // Replace Helmet tags with Head tags
    content = content.replace(/<Helmet>/g, '<Head>');
    content = content.replace(/<\/Helmet>/g, '</Head>');
    
    // Write the updated content back to the file
    fs.writeFileSync(filePath, content);
    console.log(`Updated: ${filePath}`);
  } catch (error) {
    console.error(`Error updating ${filePath}:`, error.message);
  }
}

function findAndUpdateFiles(dir) {
  const files = fs.readdirSync(dir);
  
  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      // Skip certain directories
      if (!['node_modules', '.next', 'dist', 'corrupted-src-backup', 'backup', 'ai-pages-disabled'].includes(file)) {
        findAndUpdateFiles(filePath);
      }
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      // Check if file contains react-helmet-async
      const content = fs.readFileSync(filePath, 'utf8');
      if (content.includes('react-helmet-async') || content.includes('<Helmet>')) {
        updateFile(filePath);
      }
    }
  }
}

// Start from the app directory
findAndUpdateFiles('./app');
console.log('Helmet imports updated successfully!');